src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -26,5 +32,72 @@ private static bool GetSwitchValue(string switchName, string envVariable)
 
             return ret;
         }
+
+        private static bool TryGetStringValue(string switchName, string envVariable, [NotNullWhen(true)] out string? value)
+        {
+            value = AppContext.GetData(switchName) as string;
+            if (string.IsNullOrEmpty(value))
+            {
+                value = Environment.GetEnvironmentVariable(envVariable);
+                if (string.IsNullOrEmpty(value))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static void LoadAppLocalIcu(string icuSuffixAndVersion, bool suffixWithSeparator = false)
+        {
+            ReadOnlySpan<char> icuSuffix = default;
+            ReadOnlySpan<char> version = default;
+
+            // Custom built ICU can have a suffix on the name, i.e: libicuucmyapp.so.67.1
+            // So users would set the runtime switch as: myapp:67.1
+            int indexOfSeparator = icuSuffixAndVersion.IndexOf(':');
+            if (indexOfSeparator >= 0)
+            {
+                icuSuffix = icuSuffixAndVersion.AsSpan().Slice(0, indexOfSeparator);
+
+                if (icuSuffix.Length > 35)
```

Comment:
These length checks protect buffer overruns in the unmanaged code. It may be better to do them in the unmanaged code so that the magic constants are not duplicated between the two places, and so that what check and the code that depends on this check are closed together.

Reply:
Makes sense to avoid having duplication. The reason why I added validation here was to not even bother trying to load ICU if the input was outside of the sizes we expected... I guess it wouldn’t hurt to try and load icu and if for some reason we where able to load icu and get to native code then validate that there.

Reply:
Fixed, thanks.